crates/prover-tracer/src/lib.rs

```diff
@@ -0,0 +1,283 @@
+use std::{fmt::Display, path::PathBuf, time::Duration};
+
+use opentelemetry::{global, trace::TracerProvider, KeyValue};
+use opentelemetry_otlp::WithExportConfig;
+use opentelemetry_sdk::{
+    propagation::TraceContextPropagator,
+    trace::{BatchConfigBuilder, BatchSpanProcessor, Sampler, SpanLimits},
+    Resource,
+};
+use serde::{Deserialize, Deserializer, Serialize};
+use tracing_subscriber::{
+    fmt::writer::BoxMakeWriter, prelude::*, util::SubscriberInitExt, EnvFilter,
+};
+
+/// The tracing format.
+#[derive(Serialize, Deserialize, Debug, Default, Clone, Copy, PartialEq, Eq)]
+#[serde(rename_all = "lowercase")]
+pub enum TracingFormat {
+    #[default]
+    Pretty,
+    Json,
+}
+
+/// The tracing configuration.
+#[derive(Serialize, Deserialize, Debug, Clone, Default, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+pub struct TracingConfig {
+    /// The `RUST_LOG` environment variable will take precedence over the
+    /// configuration tracing level.
+    #[serde(default)]
+    pub level: TracingLevel,
+    #[serde(default)]
+    pub outputs: Vec<TracingOutput>,
+    #[serde(default)]
+    pub format: TracingFormat,
+    /// Socket of the open telemetry agent endpoint.
+    /// If not provided open telemetry will not be used.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub otlp_agent: Option<String>,
+    /// Otlp service name.
+    /// If not provided open telemetry will not be used.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub otlp_service_name: Option<String>,
+}
+
+/// The tracing level.
+#[derive(Serialize, Deserialize, Debug, Default, Clone, Copy, PartialEq, Eq)]
+#[serde(rename_all = "lowercase")]
+pub enum TracingLevel {
+    Trace,
+    Debug,
+    #[default]
+    Info,
+    Warn,
+    Error,
+    Fatal,
+}
+
+impl Display for TracingLevel {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let level = match self {
+            TracingLevel::Trace => "trace",
+            TracingLevel::Debug => "debug",
+            TracingLevel::Info => "info",
+            TracingLevel::Warn => "warn",
+            TracingLevel::Error => "error",
+            TracingLevel::Fatal => "fatal",
+        };
+
+        write!(f, "{level}")
+    }
+}
+
+impl From<TracingLevel> for EnvFilter {
+    fn from(value: TracingLevel) -> Self {
+        EnvFilter::new(format!(
+            "warn,prover={value},aggkit={value},agglayer={value},pessimistic_proof={value}"
```

Comment:
Where are the `prover=`, `aggkit=` etc. keys coming from? I assumed these are supposed to be full crate names but might be wrong.